### ISSUE TITLE
screenMatchesGolden sometimes creates failure folders even when the test succeeds

### ISSUE THREAD

**Issue by kuhnroyal** [eBay/flutter_glove_box#36]
This seems to be caused by `await _primeImages(fileName, actualFinder);`.
Sometimes things are not completely settled here and a failure folder is created.
The test ist still successful but failure artifacts are published in the CI system.

I can work around it by using `pumpAndSettle` in the `deviceSetup` but it seems weird.

Why do the golden images needed to be primed at all? Commenting this out doesn't seem to have an effect on duration or test results.

Thanks for the library!

**Comment by coreysprague** [comment]
Thanks for submitting the issue. 

The _primeImages is a bit of hack to work around a known race condition in how images are decoded. In some cases, if this hack is omitted, the images will not be displayed. 

That being said, I think the implementation of this hack is wrong. I believe the same goal can be achieved without re-running the golden assertion. We'll take a look!

**Comment by aanelson** [comment]
This flutter thread has alternative ways we can look into doing this.
https://github.com/flutter/flutter/pull/38923#issuecomment-604030753

**Comment by christian-muertz** [comment]
How about doing sth. like this?

```
Future<void> waitForAllImages(WidgetTester tester) async {
  final Iterable<Element> imageElements = find.byType(Image).evaluate();
  await tester.runAsync(() async {
    for (Element imageElement in imageElements) {
      await precacheImage((imageElement.widget as Image).image, imageElement);
    }
  });
}
```

This resolves all ImageProviders of Image widgets and caches the result so it is available instantly afterwards.

This function works only for waiting for Image widgets. There are probably edge cases where other assets are loaded asyn.

**Comment by coreysprague** [comment]
@kuhnroyal this is fixed in 0.5.0 which should be published today or tomorrow!
